src/main/java/com/leverx/learningmanagementsystem/service/impl/CourseSettingsServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+import com.leverx.learningmanagementsystem.repository.CourseSettingsRepository;
+import com.leverx.learningmanagementsystem.service.CourseSettingsService;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.UUID;
+
+import static com.leverx.learningmanagementsystem.utils.DataFormatUtils.DATE_TIME_FORMAT;
+
+@Slf4j
+@Service
+@AllArgsConstructor
+public class CourseSettingsServiceImpl implements CourseSettingsService {
+
+    private final CourseSettingsRepository courseSettingsRepository;
+    private final CourseSettingsMapper courseSettingsMapper;
+
+    @Override
+    public CourseSettings getById(UUID id) {
+        log.info("Get course settings by id: {}", id);
+        return courseSettingsRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Course settings with id = " + id + " not found"));
+    }
+
+    @Override
+    public List<CourseSettings> getAll() {
+        log.info("Get all course settings");
+        return courseSettingsRepository.findAll();
+    }
+
+    @Override
+    @Transactional
+    public CourseSettings create(CreateCourseSettingsDto createCourseDto) {
+        CourseSettings courseSettings = courseSettingsMapper.toCourseSettings(createCourseDto);
+
+        checkDate(courseSettings);
+
+        log.info("Create course settings: {}", courseSettings);
+        courseSettingsRepository.save(courseSettings);
+
+        return courseSettings;
+    }
+
+    @Override
+    @Transactional
+    public CourseSettings update(UUID id, CreateCourseSettingsDto updateCourseDto) {
+
+        if (courseSettingsRepository.findById(id).isEmpty()) {
+            throw new EntityNotFoundException("Course settings with id = " + id + " not found");
+        }
+
+        CourseSettings courseSettings = courseSettingsMapper.toCourseSettings(updateCourseDto);
+        courseSettings.setId(id);
+
+        checkDate(courseSettings);
+
+        log.info("Update course settings: {}", courseSettings);
+        courseSettingsRepository.save(courseSettings);
+
+        return courseSettings;
+    }
+
+    @Override
+    public void delete(UUID id) {
+        getById(id);
+        log.info("Delete course settings by id: {}", id);
+        courseSettingsRepository.deleteById(id);
+    }
+
+    private void checkDate(CourseSettings courseSettings) {
+
+        if (courseSettings.getStartDate() == null || courseSettings.getEndDate() == null) {
+            throw new InvalidCourseDatesException(
+                    "Invalid date format, expected format: " + DATE_TIME_FORMAT);
+        }
+
+        if (courseSettings.getStartDate().isAfter(courseSettings.getEndDate())) {
+            throw new InvalidCourseDatesException(
+                    "Course start date is after course end date");
+        }
+    }
```

Comment:
it's better to create dedicated `CourseSettingsValidator` class for this goal. Right now, name of this method is to generic and it's better to explain what exactly you are checking (e.g. validateStartAndEndDates)